### ISSUE TITLE
Registration problem for accounts not activated

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/iRecord#590]
If an account is registered but not activated, the user is unable to change their password.  Here is an example of reports from users who get stuck in a loop:

_I'm trying to submit records and have an account but forgotten password. The reset password option sends me to a page requesting my password which I've forgotten. Now when I click the forgotten password option again it doesn't recognise my user name and when I try to set up a new account it says my email is already in use_

There are number of iRecord accounts that are not Active

**Comment by DavidRoy** [comment]
Further complaints about registration process on the iRecord forum (copied below).

I've marked this as urgent as it leads to unnecessary support time to resolve.

https://www.brc.ac.uk/irecord/node/7742#new
_I have just managed to register after about 6 tries.. I have never come across a more convoluted or confusing system. 

I am sure it will put many people off. 

On the first registering form there is nowhere to set your password... And it doesn't say that a link will be sent to you.

Then on finding the email the first link isn't to the password page.... So it's easy to go to the wrong one. 

On setting your password there isn't an obvious way to complete the page and go elsewhere._

**Comment by johnvanbreda** [comment]
Login process changed using LoginToboggan so you are now asked for your password on initial sign up. I've altered the registration email address and tweaked a few things to make it work smoothly.

I've also moved the user account page around a little so the flow is more obvious on sign up. This also involved moving the Training mode checkbox (as it didn't really make sense hidden under "What do you record") so the user guide will need a small update.

**Comment by DavidRoy** [comment]
Thanks.  I've tested and is much improved.

**Comment by kitenetter** [comment]
There have been a couple more reports in the iRecord inbox today from users who have got stuck in a similar loop. I won't re-open it at the moment, but will monitor whether this persists.

**Comment by kitenetter** [comment]
Actually one of today's accounts was the second example I've seen recently of an apparently legitimate user having their account flagged as blocked. What might cause an account to get flagged as blocked?

**Comment by DavidRoy** [comment]
@kitenetter.  Are we still getting this issue?  There are emails in the iRecord inbox to suggest there is?

**Comment by kitenetter** [comment]
Last time I was in the inbox (which was some time ago) there were certainly instances of users being unable to submit records from the apps (I think mostly from the butterflies app although people don't always say which app they are using). But I'm not sure if that is the same problem - in at least some cases problems seem to have arisen from people using the apps without first logging in.

I don't recall seeing a blocked account recently.

**Comment by kitenetter** [comment]
This problem seems to be on the rise again. One of today's emails provided this information about the sequence of events that led to the loop.

The initial problem was reported as:
"I followed the links as described to reset my forgotten password but it comes back with account not activated or blocked.  If I try to register again it says my account already exists. If I try to reset the password it says the email address doesn’t exist.  I think gremlins are in the system, how can my address be already registered but be unrecognized, not activated and blocked?"

The account was showing as blocked, so I made it active and set a temporary password. I asked what app or web page he had used:
"I started by trying to login to the Asian Hornet Watch app.  I used the website links that were emailed to me for forgotten password initially using my phone’s browser and then when this failed tried my laptop before emailing you."

I'm still unclear as to whether the problem arises from the app (Asian Hornet in this case), from trying to deal with the reset links on a mobile device, or from some other issue.
